integration-tests/contract-permits-set.spec.ts

```diff
@@ -0,0 +1,141 @@
+import { CONFIGS } from "./config";
+import { MichelsonMap, MichelCodecPacker } from "@taquito/taquito";
+import { permit_admin_42_set} from "./data/permit_admin_42_set";
+import { Parser } from '@taquito/michel-codec'
+import { importKey } from '@taquito/signer';
+import { buf2hex, hex2buf } from "@taquito/utils";
+
+const blake = require('blakejs');
+
+CONFIGS().forEach(({ lib, rpc, setup }) => {
+  const Tezos = lib; 
+  Tezos.setPackerProvider(new MichelCodecPacker());
+  const bob_address = 'tz1Xk7HkSwHv6dTEgR7E2WC2yFj4cyyuj2Gh';
+ 
+  const errors_to_missigned_bytes = (errors: any[]) => {
+    const errors_with = errors.map(x => x.with).filter(x => x !== undefined);
+    if (errors_with.length != 1){
+      throw ['errors_to_missigned_bytes: expected one error to fail "with" michelson, but found:', errors_with]
+    } else {
+      const error_with = errors_with[0];
+      const p = new Parser();
+      const michelsonCode = p.parseJSON(error_with);
+      if (error_with.prim !== 'Pair'){
+        throw ['errors_to_missigned_bytes: expected a "Pair", but found:', error_with.prim]
+      } else {
+        const error_with_args = error_with.args;
+        if (error_with_args.length !== 2){
+          throw ['errors_to_missigned_bytes: expected two arguments to "Pair", but found:', error_with_args]
+        } else {
+          if (error_with_args[0].string !== 'missigned'){
+            throw ['errors_to_missigned_bytes: expected a "missigned" annotation, but found:', error_with_args[0]]
+          } else {
+            if (typeof error_with_args[1].bytes !== 'string'){
+              throw ['errors_to_missigned_bytes: expected bytes, but found:', error_with_args[1].bytes]
+            } else {
+              return error_with_args[1].bytes
+            }
+          }
+        }
+      }
+    }
+  } 
+
+  describe(`Tests of contracts having a permit for tzip-17: ${rpc}`, () => {
+
+    beforeEach(async (done) => {
+      await setup()
+      await importKey(
```

Comment:
Running the integration tests with a faucet key can leads to `counter already in use` error if we have more than 1 PRs that run the tests at the same time. 
I suggest you to remove the importKey function and use a key from the keygen by doing `await setup(true)`

Reply:
made the change, but had to include the faucet for the test that gets the 'expected_param_hash'